cub/test/catch2_test_thread_scan_inclusive_partial.cu

```diff
@@ -0,0 +1,665 @@
+      break;
+    case 6:
+      thread_scan_inclusive_partial_kernel<6><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 7:
+      thread_scan_inclusive_partial_kernel<7><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 8:
+      thread_scan_inclusive_partial_kernel<8><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 9:
+      thread_scan_inclusive_partial_kernel<9><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 10:
+      thread_scan_inclusive_partial_kernel<10><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 11:
+      thread_scan_inclusive_partial_kernel<11><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 12:
+      thread_scan_inclusive_partial_kernel<12><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 13:
+      thread_scan_inclusive_partial_kernel<13><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 14:
+      thread_scan_inclusive_partial_kernel<14><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 15:
+      thread_scan_inclusive_partial_kernel<15><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    case 16:
+      thread_scan_inclusive_partial_kernel<16><<<1, 1>>>(d_in, d_out, scan_operator, valid_items, prefix, apply_prefix);
+      break;
+    default:
+      FAIL("Unsupported number of items");
+  }
+  REQUIRE(cudaSuccess == cudaPeekAtLastError());
+  REQUIRE(cudaSuccess == cudaDeviceSynchronize());
+}
+
+/***********************************************************************************************************************
+ * Test cases
+ **********************************************************************************************************************/
+
+C2H_TEST("ThreadScanInclusive Integral Type Tests", "[scan][thread]", integral_type_list, cub_operator_integral_list)
+{
+  using params                     = params_t<TestType>;
+  using value_t                    = typename params::item_t;
+  using output_t                   = typename params::output_t;
+  using op_t                       = c2h::get<1, TestType>;
+  using accum_t                    = ::cuda::std::__accumulator_t<op_t, value_t>;
+  using dist_param                 = dist_interval<value_t, output_t, op_t>;
+  constexpr auto scan_op           = op_t{};
+  constexpr auto operator_identity = cub_operator_to_identity<accum_t, op_t>::value();
+  const int num_items              = GENERATE_COPY(take(3, random(1, max_size)));
+  const int valid_items            = GENERATE_COPY(
+    take(1, random(2, num_items - 2)),
```

Comment:
This may potentially be an empty range, e.g., if `num_items=1`. I am not sure this is well-defined in such a case? We may want to swap the lower bound with the upper bound if that's the case.

Note for C++ `uniform_int_distribution` [ctor](https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution/uniform_int_distribution.html), it states:
> The behavior is undefined if `a > b`.